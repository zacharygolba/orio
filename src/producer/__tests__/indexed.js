// @flow

import IndexedProducer from '../indexed'

describe('IndexedProducer', () => {
  describe('source array', () => {
    let source
    let producer

    beforeEach(() => {
      source = [1, 2, 3]
      producer = new IndexedProducer(source)
    })

    test('#constructor()', () => {
      expect(producer).toMatchSnapshot()
    })

    test('#@@iterator()', () => {
      for (const item of producer) {
        expect(item).toMatchSnapshot()
      }
    })
  })

  describe('source string', () => {
    let source
    let producer

    beforeEach(() => {
      source = 'test'
      producer = new IndexedProducer(source)
    })

    test('#constructor()', () => {
      expect(producer).toMatchSnapshot()
    })

    test('#@@iterator()', () => {
      for (const item of producer) {
        expect(item).toMatchSnapshot()
      }
    })
  })
})
