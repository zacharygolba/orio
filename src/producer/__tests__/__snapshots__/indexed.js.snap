// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexedProducer source array #@@iterator() 1`] = `1`;

exports[`IndexedProducer source array #@@iterator() 2`] = `2`;

exports[`IndexedProducer source array #@@iterator() 3`] = `3`;

exports[`IndexedProducer source array #constructor() 1`] = `
IndexedProducer {
  "index": 0,
  "source": Array [
    1,
    2,
    3,
  ],
}
`;

exports[`IndexedProducer source string #@@iterator() 1`] = `"t"`;

exports[`IndexedProducer source string #@@iterator() 2`] = `"e"`;

exports[`IndexedProducer source string #@@iterator() 3`] = `"s"`;

exports[`IndexedProducer source string #@@iterator() 4`] = `"t"`;

exports[`IndexedProducer source string #constructor() 1`] = `
IndexedProducer {
  "index": 0,
  "source": "test",
}
`;
